src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/Router.js

```diff
@@ -163,6 +164,10 @@ export default class Router {
         this.handleNavigation(name, attributes, this.redirect);
     };
 
+    @action setContext = (name: string, value: any): void => {
+        this.context[name] = value;
```

Comment:
I don't think that I would add another feature to the router, that makes stuff even more complicated than it already is. Especially since there is already a `getDerivedRouteAttributes` feature in the router, that would also be suitable to solve this problem. This is how we e.g. [already successfully set the default webspace](https://github.com/sulu/sulu/blob/edbf2d5/src/Sulu/Bundle/PageBundle/Resources/js/views/WebspaceTabs/WebspaceTabs.js#L26-L32) or [other attributes in the `List` view](https://github.com/sulu/sulu/blob/edbf2d56fd7843ee6775ee47a5a0e394bb4fe708/src/Sulu/Bundle/AdminBundle/Resources/js/views/List/List.js#L39-L55).

Reply:
@danrot the context I introduced because it did feel a little bit strange importing the `userStore` into the `Router.js` which is needed when you navigate to a route without a locale (e.g. Contact) `attributes.locale` will be empty and needed to fallback to user settings content locale `userStore.contentLocale` when navigate again to a route with a locale, but if its okay for you I can import the userStore into the Router.js.

Reply:
I don't really see how these things are connected... The fallback could still be handled by the `getDerivedRouteAttributes` (as said, that's what they are built for). And you could still use an `UpdateRouteHook` to update the `contentLocale`, although I think I would not do it the way you have handled it here. I would update the content locale manually in the `List` and `Form` views/containers (not sure currently where this is done) instead of trying to find a solution working magically. If somebody creates their own route containing a `locale` parameter used differently, that could lead to very confusing behavior, which might be very hard to debug.